Makefile.msvc

```diff
@@ -0,0 +1,108 @@
+# Makefile for "nmake", part of Microsoft Visual Studio Compiler (MSVC) on Windows
```

Comment:
I wonder if we should go ahead and switch to a platform-independent build script to avoid makefile drift. Since we already include Ruby, maybe it could be a rake task?

Reply:
Fair point re: Makefile drift. I wonder if there is a way to maintain the list of tests (and examples) another way, or just discover all of them in the test/examples directories, to avoid having to add new tests to both.

Generating Makefiles could work, but I worry that the indirection (i.e. having to go through a generation script first) isn't obvious - i.e. I'd prefer if we could find a way to avoid that extra step.

Reply:
Postgres itself started using Meson for this purpose, but that of course then introduces a build-time dependency.

I think for now I'll not adjust this since this is already a massive PR, but I think we could come back to this at a future time. One approach to solving the duplication for tests/examples specifically is to have a Makefile in the `tests` and `examples` folders that can run on either platform. The main motivation for not doing that in the root folder `Makefile` is that there are a bunch of features not supported by `nmake` that make it hard to write a portable Makefile that can do the code generation/etc conditionally.